### PR TITLE
feat(pipeline-templates): config view

### PR DESCRIPTION
**Pull request by danielpeach**
@jrsquared or @anotherchrisberry please review.

@robzienert fyi

### CODE REVIEW

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-import {SubmitButton} from 'core/modal/buttons/SubmitButton';
-import {ReactInjector, NgReact } from 'core/reactShims';
-import {SETTINGS} from 'core/config/settings';
-import {IPipelineTemplate} from 'core/pipeline/config/templates/pipelineTemplate.service';
-import {TemplateDescription} from './TemplateDescription';
-import {TemplateSelector} from './TemplateSelector';
+import { $log } from 'ngimport';
+import { IHttpPromiseCallbackArg } from 'angular';
+import { cloneDeep, uniqBy } from 'lodash';
+import { Debounce } from 'lodash-decorators';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: space between external and internal imports

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+import { Application } from 'core/application/application.model';
+import { IPipeline } from 'core/domain/IPipeline';
+import { SubmitButton } from 'core/modal/buttons/SubmitButton';
+import { ReactInjector, NgReact } from 'core/reactShims';
+import { SETTINGS } from 'core/config/settings';
+import { IPipelineTemplateConfig, IPipelineTemplate } from 'core/pipeline/config/templates/pipelineTemplate.service';
+import { TemplateDescription } from './TemplateDescription';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: space between local and core imports

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx [hunk 3]
```diff
@@ -94,31 +99,62 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
...
 
+  private submitPipelineTemplateConfig(): void {
+    const config: Partial<IPipelineTemplateConfig> = {
+      name: this.state.command.name,
+      application: this.props.application.name,
+      type: 'templatedPipeline',
+      limitConcurrent: true,
+      keepWaitingPipelines: false,
+      parallel: true,
+      executionEngine: 'v2',
```
Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: v3 is the new hotness in orca, should we change this to v3?

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx [hunk 4]
```diff
@@ -180,6 +217,23 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
...
+        loadingTemplateFromSourceError: false,
+        command: Object.assign({}, this.state.command, {template: null}),
+      });
+    };
+  }
+
+  public handleSourceUrlChange(e: React.ChangeEvent<HTMLInputElement>): void {
+    const templateSourceUrl = e.target.value;
+    this.setState({templateSourceUrl});
+    this.loadPipelineTemplateFromSource(templateSourceUrl);
```
Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: Not sure if this _will_ cause any issues, but if the `loadPipelineTemplateFromSource` isn't debounced, you'll be calling setState back to back, which can sometimes have some odd rendering effects...

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx (hunk 3 above)
Comment (robzienert) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: I'd just remove this field entirely. Orca should be the one to decide the default execution engine (which it does).

Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: even better

File: app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx (hunk 4 above)
Comment (danielpeach) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: I'm confused - `loadPipelineTemplateFromSource` is debounced.

Comment (jrsquared) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: Sorry, poor clarity (and also and a poor understanding of the default options debounce) on my part, I was using debounced as a verb at runtime. I meant that if `wait` had expired for the debounced function.  I thought `leading` was the default behavior, not `trailing`. I have much less concern for back-to-back `setState` calls with a `trailing` debounce. So nothing to see here.

Comment (danielpeach) [inline app/scripts/modules/core/src/delivery/create/CreatePipelineModal.tsx]: Sounds good - thanks.

**Pull request merged**